crates/revm/src/evm_impl.rs

```diff
@@ -139,21 +129,32 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
                 .initial_account_load(self.data.env.block.coinbase, &[], self.data.db)
                 .map_err(EVMError::Database)?;
         }
-        self.load_access_list()?;
+
+        // Load access list for berlin hardfork.
+        //
+        // Loading of accounts/storages is needed to make them hot.
+        for (address, slots) in self.data.env.tx.access_list.iter() {
+            self.data
+                .journaled_state
+                .initial_account_load(*address, slots, self.data.db)
+                .map_err(EVMError::Database)?;
+        }
```

Comment:
Can you revert this, having it inside function feels better

Reply:
The problem here was that the borrow checker was complaining that `load_access_list` borrows `self` mutably